### DISCUSSION TITLE
How to retrive the element number through which ray is passing in ray tracing module

### DISCUSSION THREAD

**Discussion by an-singh** [idaholab/moose#22893] [Q&A Modules: General] [answered]
Hello friends,
I want to do ray tracing through the geometry and retrive the list of element numbers  through which ray is passing for each ray in study.

**Comment by GiudGiud** [comment]
Hello

I dont think we have that already made, though @loganharbour might know better
I would make a new vector postprocessor that would have one vector for each ray and the vector would be filled with the element IDs.
You can look at https://mooseframework.inl.gov/moose/source/vectorpostprocessors/PerProcessorRayTracingResultsVectorPostprocessor.html
for inspiration.

Guillaume

**Comment by an-singh** [comment]
Hello GiudGiud,
How will it work with "RepeatableRayStudy" UserObject  for tracing the rays.

**Reply by GiudGiud** [reply]
You can try one of the examples in the page for the VPP
It should work in any ray tracing study.
The Ray can keep a lot of additional data that will help you backtrack which elements it crossed

In the study you will want to set `always_cache_traces= true`
https://mooseframework.inl.gov/source/userobjects/RepeatableRayStudy.html

**Comment by an-singh** [comment]
can you please specify which example is helpful in getting element ID during ray tracing ......

**Reply by GiudGiud** [reply]
as I mentioned, to the best of my knowledge this has not been done before.

**Comment by loganharbour** [comment, marked as answer]
@an-singh can you please tell me what the purpose of doing this is?

I can describe how you would go about doing this, but I would prefer to have a little more context first.

**Reply by an-singh** [reply]
I have written a code to sove neutron transport equation using method of characteristics in fortran 90. To add more flexibility and design options in terms of geometry treatment and ray tracing I want to use moose framework. It will be very helpful in my research.

**Reply by an-singh** [reply]
i am still waiting for your reply @loganharbour.

**Reply by an-singh** [reply]
It is somewhat disheartening that I have not received a response after two weeks.
